### ISSUE TITLE
False negative of no-unnecessary-qualifier

### ISSUE THREAD

**Issue by KSXGitHub** [palantir/tslint#4195]
### Bug Report

- __TSLint version__: 
- __TypeScript version__: 3.1.0
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export namespace outer {
  export type Foo<X> = X[]

  export namespace inner {
  	export type Foo = outer.Foo<number>
  }
}
```

with `tslint.json` configuration:

```json
{
	"no-unnecessary-qualifier": true
}
```

#### Actual behavior

TSLint should yield no error

#### Expected behavior

It yields error

**Comment by JoshuaKGoldberg** [comment]
Seems like a duplicate of #2413. Same for here: it's spooky that there's an `outer.Foo` and an `inner.Foo`. Although the rule should be fixed to handle these cases, we should try to avoid them as much as possible.
